compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -15780,18 +15780,35 @@ OMR::Z::TreeEvaluator::arraycmpSIMDHelper(TR::Node *node,
    TR::Register * vectorOutputReg = cg->allocateRegister(TR_VRF);
    TR::Register * resultReg = needResultReg ? cg->allocateRegister() : NULL;
 
-   // VLL uses lastByteIndexReg as the highest 0-based index to load, which is length - 1
-   generateRILInstruction(cg, TR::InstOpCode::getSubtractLogicalImmOpCode(), node, lastByteIndexReg, 1);
-   if(needResultReg && isArrayCmpLen)
-      generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, resultReg, lastByteIndexReg);
+   if(needResultReg)
+      {
+      if(isArrayCmpLen)
+         generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, resultReg, lastByteIndexReg);
+      else
+         generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, resultReg, resultReg);
+      }
 
    TR::LabelSymbol * cFlowRegionStart = generateLabelSymbol(cg);
    TR::LabelSymbol * cFlowRegionEnd = generateLabelSymbol(cg);
    TR::LabelSymbol * mismatch = generateLabelSymbol(cg);
+   TR::LabelSymbol * loopStart = generateLabelSymbol(cg);
+
+   // In certain cases we can branch directly so we must set up the correct global register dependencies
+   TR::RegisterDependencyConditions* compareTargetRDC = getGLRegDepsDependenciesFromIfNode(cg, ificmpNode);
 
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionStart);
    cFlowRegionStart->setStartInternalControlFlow();
 
+   // Short circuit if the source addresses are equal
+   if(isFoldedIf && isIfxcmpBrCondContainEqual)
```

Comment:
Source1 operand and Source2 operand will be 64 Bit address. With CLR, you are generating 32-bit comparison which would be incorrect.

Reply:
using `TR::InstOpCode::getCmpLogicalRegOpCode()` instead